core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -826,6 +830,18 @@ private DefaultFunctionResolver to_days() {
         impl(nullMissingHandling(DateTimeFunction::exprToDays), LONG, DATETIME));
   }
 
+  /**
+   * TO_SECONDS(STRING/DATE/DATETIME/TIMESTAMP). return the day number of the given date.
```

Comment:
1. Fix description
> return the **day number**
2. List `LONG`
3. Accept `TIME`
4. Don't accept `TIME`, `DATE` and `DATETIME`. Values of these types are automatically casted to `TIMESTAMP` as of #1196. See example in #1254.

Reply:
I thought I addressed this in https://github.com/Bit-Quill/opensearch-project-sql/pull/232/commits/9a2377a5b8ee4bcf5df64a5cea5bf20bd069d32b. Looking closer your point 3 and 4 conflict. Am I supposed to accept `TIME`?

Reply:
No, TIME could be casted to TIMESTAMP automatically